src/subscriber.ts

```diff
@@ -425,18 +498,42 @@ export class Subscriber extends EventEmitter {
    * Acknowledges the supplied message.
    *
    * @param {Message} message The message to acknowledge.
-   * @returns {Promise}
+   * @returns {Promise<void>}
    * @private
    */
   async ack(message: Message): Promise<void> {
     const ackTimeSeconds = (Date.now() - message.received) / 1000;
     this.updateAckDeadline(ackTimeSeconds);
 
-    this._acks.add(message);
+    // Ignore this in this version of the method (but hook then/catch
+    // to avoid unhandled exceptions).
+    const resultPromise = this._acks.add(message);
+    resultPromise.then(() => {});
+    resultPromise.catch(() => {});
+
     await this._acks.onFlush();
     this._inventory.remove(message);
   }
 
+  /**
+   * Acknowledges the supplied message, expecting a response (for exactly
+   * once subscriptions).
+   *
+   * @param {Message} message The message to acknowledge.
+   * @returns {Promise<AckResponse>}
+   * @private
+   */
+  async ackWithResponse(message: Message): Promise<AckResponse> {
+    const ackTimeSeconds = (Date.now() - message.received) / 1000;
+    this.updateAckDeadline(ackTimeSeconds);
+
+    await this._acks.add(message);
```

Comment:
Sorry my experience with JS is limited: is this a blocking call? I assume no because otherwise returning a promise wouldn't make sense, but I just want to confirm.

Reply:
The way JS handles async/await is a little confusing looking. Basically the interpreter turns any async method into a couple of blocks with continuations using then() and catch(), and so the whole method ends up returning a Promise. In this particular case, the await will cause it to return a Promise that resumes the function's execution when resolve is called.